rslib/src/import_export/gather.rs

```diff
@@ -0,0 +1,284 @@
+        deck_id: Option<DeckId>,
+        with_scheduling: bool,
+    ) -> Result<()> {
+        self.decks = col.gather_decks(deck_id)?;
+        let search = optional_deck_search(deck_id);
+        self.notes = col.gather_notes(search.clone())?;
+        self.cards = col.gather_cards(search, &self.notes, deck_id)?;
+        self.notetypes = col.gather_notetypes(&self.notes)?;
+
+        if with_scheduling {
+            self.revlog = col.gather_revlog(&self.cards)?;
+            self.decks_configs = col.gather_deck_configs(&self.decks)?;
+        } else {
+            self.remove_scheduling_information(col);
+        };
+
+        Ok(())
+    }
+
+    pub(super) fn gather_media_paths(&mut self, media_folder: &Path) {
+        let mut inserter = |name: &str| {
+            if is_local_base_name(name) {
+                self.media_paths.insert(media_folder.join(name));
+            }
+        };
+        let svg_getter = svg_getter(&self.notetypes);
+        for note in self.notes.iter() {
+            gather_media_paths_from_note(note, &mut inserter, &svg_getter);
+        }
+        for notetype in self.notetypes.iter() {
+            gather_media_paths_from_notetype(notetype, &mut inserter);
+        }
+    }
+
+    fn remove_scheduling_information(&mut self, col: &Collection) {
+        self.remove_system_tags();
+        self.reset_deck_config_ids();
+        self.reset_cards(col);
+    }
+
+    fn remove_system_tags(&mut self) {
+        // TODO: case folding? child tags?
+        for note in self.notes.iter_mut() {
+            note.tags = std::mem::take(&mut note.tags)
+                .into_iter()
+                .filter(|tag| !matches!(tag.as_str(), "marked" | "leech"))
+                .collect();
+        }
+    }
+
+    fn reset_deck_config_ids(&mut self) {
+        for deck in self.decks.iter_mut() {
+            if let Ok(normal_mut) = deck.normal_mut() {
+                normal_mut.config_id = 1;
+            } else {
+                // TODO: scheduling case
+                deck.kind = DeckKind::Normal(NormalDeck {
+                    config_id: 1,
+                    ..Default::default()
+                })
+            }
+        }
+    }
+
+    fn reset_cards(&mut self, col: &Collection) {
+        let mut position = col.get_next_card_position();
+        for card in self.cards.iter_mut() {
+            if card.ctype != CardType::New || card.queue != CardQueue::New {
+                card.due = card.original_position.unwrap_or_else(|| {
+                    position += 1;
+                    position - 1
+                }) as i32;
+            }
+            card.interval = 0;
+            card.ease_factor = 0;
+            card.reps = 0;
+            card.lapses = 0;
+            card.original_deck_id = DeckId(0);
+            card.original_due = 0;
+            card.original_position = None;
+            card.queue = CardQueue::New;
+            card.ctype = CardType::New;
+            card.flags = 0;
```

Comment:
Just noticed an unhandled edge case: If a card from a filtered deck is rescheduled as new, shouldn't it's original position (odue) also be reused if it's of type new?

Reply:
Hm, even if a card is new and in a regular deck, it's assigned a new position. Should I make a PR that can be merged into 2.1.50?

Reply:
For our export case, it sounds like we should be doing that. In your follow-up comment, are you saying that reschedule_cards_as_new() assigns new positions if there is no original position or restore_position is false? I thought that was intended behaviour, so that users could send new cards to the back of the queue. Have I missed something?

Reply:
I'm only talking about the `restore_position == true` case here. In that case, I wouldn't expect the current position of new cards to change. Rather something like the following should be used:

```rs
    fn last_position(&self: Card) -> Option<u32> {
        if self.ctype == CardType::New {
            Some(if self.original_deck_id.0 != 0 {
                self.original_due
            } else {
                self.due
            } as u32)
        } else {
            self.original_position
        }
    }
```

Reply:
Ok, I think I get you - you're saying you think the behaviour should be 'restore original position if available (otherwise, leave current position alone if it's already new)', instead of '(otherwise, place card at end of queue)'? That does seem to make more sense; a PR would be great.